firmware/octopi_firmware_v2/main_controller_teensy41/main_controller_teensy41.ino

```diff
@@ -1500,32 +1604,70 @@ void loop() {
             stage_PID_enabled[axis] = 0;
             break;
           }
+        case INITFILTERWHEEL:
```

Comment:
Ah I see, we have a separate filter wheel initialization so the `STAGE_AXIS` loops were probably correct.  You can ignore those if so, but since we want them separate it may make sense to have it be harder to accidentally use the W index of the tmc4361* arrays accidentally when it isn't initialized.  For instance, having a `primary_axes_motors[STAGE_AXES]` array and a separate `w_axis_motor` variable instead of a single `tmc4361*` array.

Also, it looks like we allow W axis init via `INITIALIZE` as well.  Do we want that?  Or should there only be 1 path to initialize each of the axes?

As is, we partially initialize some axes in `setup`, initialize the 3 primary motion axes and maybe the W axis in `INITIALIZE`, and initialize the W axis in `INITFILTERWHEEL`.  This means if we make changes we need to keep all 3 of those in sync, which is tricky!  It'd be better if it was all in one place.  As a for instance, we could have an `ENABLE_FILTER_WHEEL` command that sets `enable_filterwheel = true`, then do all initialization in `INITIALIZE` (and delete the initialization bits in `setup` since those are partial anyway).  

There are a bunch of possible ways, but eliminating the duplicated logic and simplifying the number of different paths we need to test and be aware of are the goals.  However we do that!

Reply:
Agree we should have only one path to initialize each axis. Let's only initialize the first 3 in `INITIALIZE` before we do more code refactoring. @veerwang

Reply:
And we may set `N_MOTOR` to 4 in during `INITFILTERWHEEL`

Reply:
Sure, I agree with you. I would remove W axis related init in INITAILIZE in to INITFILTERWHEEL.